scrapinghub/client/utils.py

```diff
@@ -133,35 +125,33 @@ def list(self, *args, **kwargs):
         return list(self.iter(*args, **kwargs))
 
 
-class _MappingProxy(_ProxyBase):
-
-    def __init__(self, cls, client, key):
-        super(_MappingProxy, self).__init__(cls, client, key)
+class _MappingProxy(_Proxy):
 
-        # cls_methods methods will be imported from origin class
-        origin_cls_methods = ['__iter__', '__len__', '__str__', '__repr__',
-                              '__getitem__', '__setitem__', '__delitem__']
+    def count(self):
+        return len(next(self._origin.apiget()))
 
-        other_attrs = ['ignore_fields', 'expire', '_data', '_deleted',
-                       'save', 'liveget', ('iterkeys', '__iter__')]
+    def get(self, key):
+        return next(self._origin.apiget(key))
 
-        # blacklisted attributes should be just skipped on proxying
-        blacklisted_attrs = ['__class__', '__init__', '__str__', '__repr__']
+    def set(self, key_or_values, value=None):
+        if isinstance(key_or_values, six.string_types):
+            self._origin.apipost(key_or_values,
+                                 data=json.dumps(value),
+                                 is_idempotent=True)
+        elif isinstance(key_or_values, dict):
+            data = next(self._origin.apiget())
+            data.update(key_or_values)
```

Comment:
@vshlapakov that's exactly what I proposed above, sorry if it wasn't clear enough :)